tests/test_cases/test_cocotb/test_timing_triggers.py

```diff
@@ -244,3 +247,16 @@ async def test_singleton_isinstance(dut):
     assert isinstance(NextTimeStep(), NextTimeStep)
     assert isinstance(ReadOnly(), ReadOnly)
     assert isinstance(ReadWrite(), ReadWrite)
+
+
+@cocotb.test()
+async def test_neg_timer(dut):
+    """Test negative timer values are forbidden"""
+    with assert_raises(TriggerException):
+        Timer(-42)  # no need to even `await`, constructing it is an error
+    # handle 0 special case
+    with warnings.catch_warnings(record=True) as w:
+        Timer(0)
+        assert "Timer setup with value 0, which might exhibit undefined behavior in some simulators" in str(w[-1].message)
+        assert issubclass(w[-1].category, RuntimeWarning)
+        warnings.simplefilter("ignore")
```

Comment:
```suggestion
    with warnings.catch_warnings(record=True) as w:
        warnings.simplefilter("always")
        Timer(0)
        assert "Timer setup with value 0, which might exhibit undefined behavior in some simulators" in str(w[-1].message)
        assert issubclass(w[-1].category, RuntimeWarning)
```

Reply:
Sorry for the pain here - this is exactly why I would prefer to use `with pytest.warns` rather than rolling our own!

Reply:
:D yes, because the other version was also fail
![grafik](https://user-images.githubusercontent.com/25175069/95518663-d31b8180-09c3-11eb-93bf-93523cb3223d.png)